### DISCUSSION TITLE
Observables API

### DISCUSSION THREAD

**Discussion by olierxleben** [microsoft/devicescript#479] [Q&A] [answered]
Hi, 

coming from Angular land I felt very at home with observables and I am already very pleased with device script platform.
I wondered about the subset you provide and do you plan to extend the observable operators in the future? I was wondering about switchMap in particular, because it is a very common operator for changing sources. 

Thanks

**Comment by pelikhan** [comment]
This is great feedback. We did not plan to extend it ... ourselves but we are definitely open for contributions.

To be perfectly honest, we wanted to start with a smallish set of operators and discover the missing ones on the go.

**Reply by pelikhan** [reply, marked as answer]
Started on switchMap, https://github.com/microsoft/devicescript/pull/508
